pkg/kp/consulutil/watch.go

```diff
@@ -208,3 +209,102 @@ func WatchNewKeys(pairsChan <-chan api.KVPairs, onNewKey NewKeyHandler, done <-c
 		}
 	}
 }
+
+type WatchedChanges struct {
+	Created api.KVPairs
+	Updated api.KVPairs
+	Deleted api.KVPairs
+}
+
+// WatchDiff watches a Consul prefix for changes and categorizes them
+// into create, update, and delete, please note that if a kvPair was
+// create and modified before this starts watching, this watch will
+// treat it as a create
+func WatchDiff(
+	prefix string,
+	clientKV ConsulLister,
+	quitCh <-chan struct{},
+	outErrors chan<- error,
+) <-chan *WatchedChanges {
+	outCh := make(chan *WatchedChanges)
```

Comment:
We've been trying to follow the pattern of: those who open channels are responsible for ensuring that they are closed. This might mean that this function should instead take a (chan *WatchedChanges) as an argument vs returning one to the caller.